.github/workflows/sync-linguist.yml

```diff
@@ -0,0 +1,109 @@
+        git checkout -b $branch_name
+        echo "::set-output name=branch_name::$branch_name"
+    - uses: actions/checkout@v2
+      with:
+        repository: github/linguist
+        path: .linguist
+        fetch-depth: 0
+    - name: check out latest release
+      id: linguist-release
+      # the `grep -v "-"` is to exclude any pre-release versions.
+      # Linguist doesn't have any right now, but just in case.
+      run: |
+        set -euo pipefail
+        IFS=$'\n\t'        
+        cd .linguist
+        if [[ -n "${{ github.event.inputs.linguist_tag }}" ]]; then
+          echo "Using tag override '${{ github.event.inputs.linguist_tag }}'"
+          latest="${{ github.event.inputs.linguist_tag }}"
+        else
+          latest=$(git tag --list | \
+                     grep -v "-" | \
+                     sort --version-sort --reverse | \
+                     head -1)
+        fi
+
+        if [[ -z "$latest" ]]; then
+          echo "could not determine latest Linguist version"
+          exit 1
+        fi
+
+        echo "::set-output name=linguist_version::$latest"
+        git checkout $latest
+      
+        commit=$(git rev-parse HEAD)
+        if [[ -z "$commit" ]]; then
+          echo "could not determine latest Linguist commit"
+          exit 1
+        fi
+
+        echo "::set-output name=linguist_commit::$commit"
+
+        cd ..
+    - name: Update Linguist commit
+      run: |
+        sed --in-place --regexp-extended 's/(commit[[:space:]]+=[[:space:]])("[a-f0-9]{40}")/\1"${{ steps.linguist-release.outputs.linguist_commit }}"/' internal/code-generator/generator/generator_test.go
+    - name: Generate code
+      run: make code-generate
+    - name: Commit changes
+      id: commit
+      run: |
+        set -euo pipefail
+        IFS=$'\n\t'
+        echo "git current state:"
+        git status
+
+        if [[ -n "$(git status --porcelain)" ]]; then
+          echo "Creating Linguist update commit"
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add .
+          git commit -m "Updated Linguist to ${{ steps.linguist-release.outputs.linguist_version }}"
+          git push --set-upstream origin ${{ steps.branch.outputs.branch_name }}
+          echo "Changes committed. Will create PR."
+          echo "::set-output name=needs_pr::true"
+          exit 0
+        fi
+          
+        echo "Linguist update unncessary"
+        echo "::set-output name=needs_pr::false"
+    - name: Create Pull Request
+      id: open-pr
+      uses: repo-sync/pull-request@v2
+      if: ${{ steps.commit.outputs.needs_pr == 'true' }}
+      with:
+        source_branch: ${{ steps.branch.outputs.branch_name }}
+        pr_title: "Update Linguist to ${{ steps.linguist-release.outputs.linguist_version }}"
+        pr_body: "Automated Linguist update :robot:"
+        destination_branch: "master"
+        github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Every repository has a `secrets.GITHUB_TOKEN` already defined; I didn't have to add anything. This _might_ not be true for scheduled Actions, but I didn't have a way to test it. If the scheduled Action starts failing after you merge it, I'll help debug. If you _do_ need to define a secret, it will likely require `repo:public_repo` permissions, because it commits code and creates PRs.

Your question about permissions reminds me that [workflows themselves can have permissions](https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/) and I will add them to this workflow.

Reply:
Got it, and thank you for brining workflow permissions up!